lib/streamlit/runtime/context.py

```diff
@@ -307,3 +307,17 @@ def url(self) -> str | None:
         if ctx is None or ctx.context_info is None:
             return None
         return ctx.context_info.url
+
+    @property
+    @gather_metrics("context.ip_address")
+    def ip_address(self) -> str | None:
+        """The read-only IP address of the user's connection.
+        This should not be used for security measures as it can be easily spoofed.
```

Comment:
nit: maybe we can add a bit more technical info here where this ip is from and in which deployment scenarios its not available (e.g. in proxy setup?).

Reply:
Yeah, we will definitely do that, this is just a draft version.